api/everest.go

```diff
@@ -120,16 +131,28 @@ func (e *EverestServer) initHTTPServer() error {
 
 	// Use our validation middleware to check all requests against the OpenAPI schema.
 	apiGroup := e.echo.Group(basePath)
-	apiGroup.Use(e.authenticate)
 	apiGroup.Use(middleware.OapiRequestValidatorWithOptions(swagger, &middleware.Options{
 		SilenceServersWarning: true,
 	}))
+	apiGroup.Use(e.jwtMiddleWare())
 	apiGroup.Use(e.checkOperatorUpgradeState)
 	RegisterHandlers(apiGroup, e)
 
 	return nil
 }
 
+func (e *EverestServer) jwtMiddleWare() echo.MiddlewareFunc {
+	tokenLookup := "header:Authorization:Bearer "
+	tokenLookup = tokenLookup + ",cookie:" + common.EverestTokenCookie
+	return echojwt.WithConfig(echojwt.Config{
+		Skipper: func(c echo.Context) bool {
+			return strings.Contains(c.Request().URL.Path, "session")
+		},
+		SigningKey:  []byte(e.config.JWTSigningKey),
+		TokenLookup: tokenLookup,
+	})
+}
```

Comment:
I don't think you need to change it but I just wanted to clarify something. During the demo, you mentioned that this middleware would not only validate the authenticity of the JWT but it would also validate if the user account exists. I believe this isn't true thought, it seems to me that it only checks if the JWT was signed by everest but it doesn't check the user account at all, am I missing something?

Reply:
No, you're right. We don't check if the user exists when validating the token. I probably had this mixed up with the /session endpoint during the demo..